M/mlpack/build_tarballs.jl

```diff
@@ -0,0 +1,218 @@
+## We'll use BinaryBuilder to build the Julia bindings and the scripts that are
+## used for them.  Then, we'll set aside the built .jl files, modify them as
+## necessary, and build the mlpack.jl project and push that to Github.
+
+using BinaryBuilder
+
+# Set sources and other environment variables.
+name = "mlpack"
+version = v"3.3.0-a1"
+sources = [
+    ("http://sourceforge.net/projects/arma/files/armadillo-9.800.3.tar.xz" =>
```

Comment:
Armadillo should be built as a separate package

Reply:
Are you sure?  I am only using Armadillo headers in this package.  The Armadillo shared object, which can be built if desired, is just a wrapper around any underlying BLAS or LAPACK (in this case, OpenBLAS).  If I did make an Armadillo package, I'm not sure what its output would be, since in my view in a situation like this it's safer to link directly to OpenBLAS instead of using Armadillo's wrapper library around OpenBLAS.  Anyway, let me know what you think is best.

Reply:
I mean, having armadillo in its ow package makes the builder much cleaner, and the life easier for other users, too.

Reply:
I guess the `boost` package must be installing headers also (although nothing seems to be specified in `products` in that `build_tarballs.jl`), so I can do the same here regardless of whether downstream dependencies (like here) use it or not.  Let me get started on that and I'll open a PR shortly. :+1:

Reply:
> I guess the boost package must be installing headers also

That's what we usually do if we compile from source.  It's pretty hard to build dependencies without header files :wink: 

> (although nothing seems to be specified in products in that build_tarballs.jl)

We are mostly interested in `LibraryProduct`s and `ExecutableProduct` because they need particular care, like `dlopen`ing the libraries in the right order and setting the necessary environment variables around executables.  Declaring a `Product` also defines in the JLL package a variable with the name provided as second argument, so that we can call the library or run the executable using those names.  In the JLL package there is usually little need to reference a header file, so in general we don't put header files in the list of products, there is very little need for them and having too many variables pollutes the namespace.

Reply:
Got it, thank you for the explanation.  I just wasn't sure if I needed to explicitly reference the header files, but I guess `build_tarballs()` must be automatically collecting those.  In any case, I just opened #388 for Armadillo.